src/prefect/client/client.py

```diff
@@ -330,6 +330,18 @@ def _send_request(
             )
 
         # Check if request returned a successful status
+        if response.status_code == 400:
+            try:
+                msg = (
+                    f"400 Client Error: Bad Request for url: {url}\n\n"
+                    f"This is likely caused by a poorly formatted GraphQL query or mutation. "
+                    f"GraphQL sent:\n\n{parse_graphql(params)}"
+                )
+                raise ClientError(msg)
+            except ClientError:
+                raise
+            except Exception:
+                pass
```

Comment:
I find the logic above confusing (we try to raise an error, and ignore the error if it's raised properly). Under what reasons would the formatting fail, and if it does is there another nice error message we could give rather than the one raised by `raise_for_status()`? If not,  falling back to `response.raise_for_status()` seems fine to me - I'd find the following logic cleaner though:

```suggestion
        if response.status_code == 400:
            try:
                msg = (
                    f"400 Client Error: Bad Request for url: {url}\n\n"
                    f"This is likely caused by a poorly formatted GraphQL query or mutation. "
                    f"GraphQL sent:\n\n{parse_graphql(params)}"
                )
            except Exception:
                # failed to format, raise as normal below
                pass
            else:
                raise ClientError(msg)
```

Reply:
Makes sense - will update